packages/visual-dev/src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import styled from 'styled-components'
+
+import { custom } from './Styles.tsx'
+
+const default_size = {
+	"small": "12px",
+	"medium": "23px",
+	"large": "36px"
+}
+
+interface IconProps {
+	icon: string;
+	color?: string;
+	size?: "small" | "medium" | "large" | string;
+	padding?: string;
+	margin?: string;
+}
+
+export const StyledIcon = styled.i<{
+	color?: string;
+	size: "small" | "medium" | "large" | string;
+	padding?: string;
+	margin?: string;
+}>`
+	color: ${(props) => props.color};
+	font-size: ${(props) => default_size.hasOwnProperty(props.size) ? default_size[props.size] : props.size};
+	padding: ${(props) => props.padding};
+	margin: ${(props) => props.margin};
+`;
+
+export const StyledSVG = styled.div<{
+	color: string;
+	size: "small" | "medium" | "large" | string;
+	padding: string;
+	margin: string;
```

Comment:
I understand why you've repeated the props between these three components but we should still try to avoid repeating types as much as possible. Try to only use `IconProps` for the styled-component props (might be a good chance to get into more advanced typescript if you aren't already familiar)

Reply:
Created two interfaces, `OptionProps` and `StyleProps` and combined them for the component.